diff --git a/node_modules/react-notion-x/build/index.js b/node_modules/react-notion-x/build/index.js
index a18bcfd..f3b7c7a 100644
--- a/node_modules/react-notion-x/build/index.js
+++ b/node_modules/react-notion-x/build/index.js
@@ -707,7 +707,49 @@ function SvgTypeGitHub(props) {
     fill: "#161614"
   })));
 }
+
+function SvgTypeFigma(props) {
+  return /* @__PURE__ */ React9.createElement("svg", __spreadValues({
+      viewBox: "0 0 38 57",
+      width: "32px",
+      height: "32px"
+    }),
+      /* @__PURE__ */ React9.createElement("g", null,
+        /* @__PURE__ */ React9.createElement("path",
+          {
+            d: "M19 28.5C19 25.9804 20.0009 23.5641 21.7825 21.7825C23.5641 20.0009 25.9804 19 28.5 19C31.0196 19 33.4359 20.0009 35.2175 21.7825C36.9991 23.5641 38 25.9804 38 28.5C38 31.0196 36.9991 33.4359 35.2175 35.2175C33.4359 36.9991 31.0196 38 28.5 38C25.9804 38 23.5641 36.9991 21.7825 35.2175C20.0009 33.4359 19 31.0196 19 28.5Z",
+            fill: "#1ABCFE"
+          }),
+        /* @__PURE__ */ React9.createElement("path",
+          {
+            d: "M0 47.5C0 44.9804 1.00089 42.5641 2.78249 40.7825C4.56408 39.0009 6.98044 38 9.5 38H19V47.5C19 50.0196 17.9991 52.4359 16.2175 54.2175C14.4359 55.9991 12.0196 57 9.5 57C6.98044 57 4.56408 55.9991 2.78249 54.2175C1.00089 52.4359 0 50.0196 0 47.5H0Z",
+            fill: "#0ACF83"
+          }
+        ),
+        /* @__PURE__ */ React9.createElement("path",
+          {
+            d: "M19 0V19H28.5C31.0196 19 33.4359 17.9991 35.2175 16.2175C36.9991 14.4359 38 12.0196 38 9.5C38 6.98044 36.9991 4.56408 35.2175 2.78249C33.4359 1.00089 31.0196 0 28.5 0L19 0Z",
+            fill: "#FF7262"
+          }
+        ),
+        /* @__PURE__ */ React9.createElement("path",
+          {
+            d: "M0 9.5C0 12.0196 1.00089 14.4359 2.78249 16.2175C4.56408 17.9991 6.98044 19 9.5 19H19V0H9.5C6.98044 0 4.56408 1.00089 2.78249 2.78249C1.00089 4.56408 0 6.98044 0 9.5H0Z",
+            fill: "#F24E1E"
+          }
+        ),
+        /* @__PURE__ */ React9.createElement("path",
+          {
+            d: "M0 28.5C0 31.0196 1.00089 33.4359 2.78249 35.2175C4.56408 36.9991 6.98044 38 9.5 38H19V19H9.5C6.98044 19 4.56408 20.0009 2.78249 21.7825C1.00089 23.5641 0 25.9804 0 28.5H0Z",
+            fill: "#A259FF"
+          }
+        )
+    )
+  )
+}
+
 var type_github_default = SvgTypeGitHub;
+var type_figma_default = SvgTypeFigma;
 
 // src/components/eoi.tsx
 var EOI = ({ block, inline, className }) => {
@@ -722,6 +764,7 @@ var EOI = ({ block, inline, className }) => {
   const lastUpdatedAt = (_c = attributes.find((attr) => attr.id === "updated_at")) == null ? void 0 : _c.values[0];
   const lastUpdated = lastUpdatedAt ? formatNotionDateTime(lastUpdatedAt) : null;
   let externalImage;
+  let figmaThumbnailUrl;
   switch (domain) {
     case "github.com":
       externalImage = /* @__PURE__ */ React10.createElement(type_github_default, null);
@@ -730,6 +773,10 @@ var EOI = ({ block, inline, className }) => {
         owner = parts[parts.length - 1];
       }
       break;
+    case "www.figma.com":
+      figmaThumbnailUrl = attributes.find((attr) => attr.id === "thumbnail_url")?.values[0];
+      externalImage = /* @__PURE__ */ React10.createElement(type_figma_default, null);
+      break;
     default:
       if (true) {
         console.log(
@@ -739,24 +786,40 @@ var EOI = ({ block, inline, className }) => {
       }
       return null;
   }
-  return /* @__PURE__ */ React10.createElement(components.Link, {
-    target: "_blank",
-    rel: "noopener noreferrer",
-    href: original_url,
-    className: cs(
-      "notion-external",
-      inline ? "notion-external-mention" : "notion-external-block notion-row",
-      className
+  return /* @__PURE__ */ React10.createElement("div", { className: cs("notion-eoi") },
+    /* @__PURE__ */ React10.createElement("div", { className: "notion-eoi-flex" },
+      /* @__PURE__ */ React10.createElement("div", { className: "notion-eoi-overflow" },
+        /* @__PURE__ */ React10.createElement("div", { className: "notion-eoi-position" },
+        /* @__PURE__ */ React10.createElement("div", { className: "notioni-eoi-container" },
+            figmaThumbnailUrl && /* @__PURE__ */ React10.createElement("img", {
+              src: figmaThumbnailUrl,
+              refferrerpolicy: "same-origin",
+              className: cs("notion-external-thumbnail")
+            }),
+            /* @__PURE__ */ React10.createElement(components.Link, {
+              target: "_blank",
+              rel: "noopener noreferrer",
+              href: original_url,
+              className: cs(
+                "notion-external",
+                inline ? "notion-external-mention" : "notion-external-block notion-row",
+                className
+              )
+            },
+            externalImage && /* @__PURE__ */ React10.createElement("div", {
+              className: "notion-external-image"
+            }, externalImage), /* @__PURE__ */ React10.createElement("div", {
+              className: "notion-external-description"
+            }, /* @__PURE__ */ React10.createElement("div", {
+              className: "notion-external-title"
+            }, title), (owner || lastUpdated) && /* @__PURE__ */ React10.createElement("div", {
+              className: "notion-external-subtitle"
+            }, owner && /* @__PURE__ */ React10.createElement("span", null, owner), owner && lastUpdated && /* @__PURE__ */ React10.createElement("span", null, " \u2022 "), lastUpdated && /* @__PURE__ */ React10.createElement("span", null, "Updated ", lastUpdated))))
+          )
+        )
+      )
     )
-  }, externalImage && /* @__PURE__ */ React10.createElement("div", {
-    className: "notion-external-image"
-  }, externalImage), /* @__PURE__ */ React10.createElement("div", {
-    className: "notion-external-description"
-  }, /* @__PURE__ */ React10.createElement("div", {
-    className: "notion-external-title"
-  }, title), (owner || lastUpdated) && /* @__PURE__ */ React10.createElement("div", {
-    className: "notion-external-subtitle"
-  }, owner && /* @__PURE__ */ React10.createElement("span", null, owner), owner && lastUpdated && /* @__PURE__ */ React10.createElement("span", null, " \u2022 "), lastUpdated && /* @__PURE__ */ React10.createElement("span", null, "Updated ", lastUpdated))));
+  )
 };
 
 // src/components/graceful-image.tsx
@@ -2259,7 +2322,8 @@ var Block = (props) => {
       }
       if ((_g = block.format) == null ? void 0 : _g.toggleable) {
         return /* @__PURE__ */ React28.createElement("details", {
-          className: cs("notion-toggle", blockId)
+          className: cs("notion-toggle", blockId),
+          open: true
         }, /* @__PURE__ */ React28.createElement("summary", null, headerBlock), /* @__PURE__ */ React28.createElement("div", null, children));
       } else {
         return headerBlock;
@@ -2492,7 +2556,8 @@ var Block = (props) => {
     }
     case "toggle":
       return /* @__PURE__ */ React28.createElement("details", {
-        className: cs("notion-toggle", blockId)
+        className: cs("notion-toggle", blockId),
+        open: true
       }, /* @__PURE__ */ React28.createElement("summary", null, /* @__PURE__ */ React28.createElement(Text, {
         value: (_E = block.properties) == null ? void 0 : _E.title,
         block
diff --git a/node_modules/react-notion-x/build/third-party/collection.js b/node_modules/react-notion-x/build/third-party/collection.js
index 216ef37..3c893bf 100644
--- a/node_modules/react-notion-x/build/third-party/collection.js
+++ b/node_modules/react-notion-x/build/third-party/collection.js
@@ -3507,11 +3507,7 @@ var PageTitleImpl = (_a) => {
   }
   return /* @__PURE__ */ React4.createElement("span", __spreadValues({
     className: cs("notion-page-title", className)
-  }, rest), /* @__PURE__ */ React4.createElement(PageIcon, {
-    block,
-    defaultIcon,
-    className: "notion-page-title-icon"
-  }), /* @__PURE__ */ React4.createElement("span", {
+  }, rest), /* @__PURE__ */ React4.createElement("span", {
     className: "notion-page-title-text"
   }, /* @__PURE__ */ React4.createElement(Text, {
     value: (_b2 = block.properties) == null ? void 0 : _b2.title,
@@ -6438,7 +6434,6 @@ function Table({ blockIds = [], collection, collectionView, width, padding }) {
   }
   return /* @__PURE__ */ React51.createElement("div", {
     className: "notion-table",
-    style: tableStyle
   }, /* @__PURE__ */ React51.createElement("div", {
     className: "notion-table-view",
     style: tableViewStyle
diff --git a/node_modules/react-notion-x/src/styles.css b/node_modules/react-notion-x/src/styles.css
index 71b8b4a..600a125 100644
--- a/node_modules/react-notion-x/src/styles.css
+++ b/node_modules/react-notion-x/src/styles.css
@@ -1,7 +1,7 @@
 :root {
   --notion-font: ui-sans-serif, system-ui, apple-system, BlinkMacSystemFont,
-    'Segoe UI', Helvetica, 'Apple Color Emoji', Arial, sans-serif,
-    'Segoe UI Emoji', 'Segoe UI Symbol';
+  'Segoe UI', Helvetica, 'Apple Color Emoji', Arial, sans-serif,
+  'Segoe UI Emoji', 'Segoe UI Symbol';
   --fg-color: rgb(55, 53, 47);
   --fg-color-0: rgba(55, 53, 47, 0.09);
   --fg-color-1: rgba(55, 53, 47, 0.16);
@@ -77,7 +77,7 @@
   --fg-color-6: #fff;
   --fg-color-icon: #fff;
 
-  --bg-color: #2f3437;
+  --bg-color: #191919;
   --bg-color-0: rgb(71, 76, 80);
   --bg-color-1: rgb(63, 68, 71);
   --bg-color-2: rgba(135, 131, 120, 0.15);
@@ -208,7 +208,7 @@
 }
 
 .notion-aside-table-of-contents
-  .notion-table-of-contents-item-indent-level-0:first-of-type {
+.notion-table-of-contents-item-indent-level-0:first-of-type {
   margin-top: 0;
 }
 
@@ -607,7 +607,7 @@
 
 .notion-page-icon {
   font-family: 'Apple Color Emoji', 'Segoe UI Emoji', NotoColorEmoji,
-    'Noto Color Emoji', 'Segoe UI Symbol', 'Android Emoji', EmojiSymbols;
+  'Noto Color Emoji', 'Segoe UI Symbol', 'Android Emoji', EmojiSymbols;
   font-size: 1.1em;
   fill: var(--fg-color-6);
   color: var(--fg-color-icon);
@@ -755,11 +755,13 @@ svg.notion-page-icon {
   border-radius: 3px;
   font-size: 85%;
   font-family: 'SFMono-Regular', Consolas, 'Liberation Mono', Menlo, Courier,
-    monospace;
+  monospace;
 }
 
 .notion-inline-underscore {
-  text-decoration: underline;
+  text-underline-offset: 3px;
+  text-decoration: underline solid 0.1px;
+  text-decoration-color: #9d9d9d;
 }
 
 .notion-list {
@@ -819,7 +821,7 @@ svg.notion-page-icon {
 
 .notion-asset-wrapper iframe {
   border: none;
-  background: rgb(247, 246, 245);
+  background: rgba(0, 0, 0, 0);
 }
 
 .notion-text {
@@ -856,7 +858,7 @@ svg.notion-page-icon {
   overflow: auto;
   background: var(--bg-color-1);
   font-family: SFMono-Regular, Consolas, 'Liberation Mono', Menlo, Courier,
-    monospace;
+  monospace;
 }
 
 .notion-code-copy {
@@ -876,10 +878,10 @@ svg.notion-page-icon {
   cursor: pointer;
 
   transition: background-color 0.2s cubic-bezier(0.3, 0, 0.5, 1),
-    color 0.2s cubic-bezier(0.3, 0, 0.5, 1),
-    border-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);
+  color 0.2s cubic-bezier(0.3, 0, 0.5, 1),
+  border-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);
   box-shadow: 0 1px 0 rgba(27, 31, 36, 0.04),
-    inset 0 1px 0 rgba(255, 255, 255, 0.25);
+  inset 0 1px 0 rgba(255, 255, 255, 0.25);
   background-color: #f6f8fa;
 
   color: #24292f;
@@ -1205,8 +1207,7 @@ svg.notion-page-icon {
 }
 
 .notion-table {
-  width: 100vw;
-  max-width: 100vw;
+  width: 100%;
   align-self: center;
   overflow: auto hidden;
 }
@@ -1700,14 +1701,14 @@ svg.notion-page-icon {
 }
 
 .notion-board-view-size-small
-  .notion-collection-card
-  .notion-collection-card-cover {
+.notion-collection-card
+.notion-collection-card-cover {
   height: 100px;
 }
 
 .notion-board-view-size-large
-  .notion-collection-card
-  .notion-collection-card-cover {
+.notion-collection-card
+.notion-collection-card-cover {
   height: 180px;
 }
 
@@ -2027,8 +2028,8 @@ svg.notion-page-icon {
 }
 
 .notion-collection-card-property
-  .notion-property-relation
-  .notion-page-title-text {
+.notion-property-relation
+.notion-page-title-text {
   border-bottom: 1px solid;
 }
 
@@ -2097,8 +2098,8 @@ svg.notion-page-icon {
 }
 
 .notion-collection-row-property
-  .notion-property-relation
-  .notion-page-title-text {
+.notion-property-relation
+.notion-page-title-text {
   border-bottom: 1px solid;
 }
 
@@ -2191,7 +2192,7 @@ svg.notion-page-icon {
 
 .lazy-image-real.medium-zoom-image {
   transition: transform 0.3s cubic-bezier(0.2, 0, 0.2, 1),
-    opacity 400ms ease-out !important;
+  opacity 400ms ease-out !important;
   will-change: opacity, transform;
 }
 
@@ -2374,7 +2375,7 @@ svg.notion-page-icon {
 
 .notion-search {
   box-shadow: rgba(15, 15, 15, 0.05) 0px 0px 0px 1px,
-    rgba(15, 15, 15, 0.1) 0px 5px 10px, rgba(15, 15, 15, 0.2) 0px 15px 40px;
+  rgba(15, 15, 15, 0.1) 0px 5px 10px, rgba(15, 15, 15, 0.2) 0px 15px 40px;
   border-radius: 3px;
   background: var(--bg-color);
 
@@ -2734,7 +2735,7 @@ svg.notion-page-icon {
   border-radius: 6;
   padding: 0;
   box-shadow: 0px 10px 38px -10px rgba(22, 23, 24, 0.35),
-    0px 10px 20px -15px rgba(22, 23, 24, 0.2);
+  0px 10px 20px -15px rgba(22, 23, 24, 0.2);
   animation-duration: 400ms;
   animation-timing-function: cubic-bezier(0.16, 1, 0.3, 1);
   animation-fill-mode: forwards;
@@ -2801,3 +2802,36 @@ svg.notion-page-icon {
   text-decoration: underline;
   cursor: pointer;
 }
+
+.notion-external-thumbnail {
+  display: block;
+  object-fit: cover;
+  border-radius: 1px;
+  pointer-events: auto;
+  width: 100%;
+}
+
+.notion-eoi {
+  overflow: hidden;
+  border-radius: 3px 3px 0px 0px;
+  position: relative;
+}
+
+.notion-eoi-flex {
+  display: flex;
+}
+
+.notion-eoi-overflow {
+  position: relative;
+  overflow: hidden;
+  flex-grow: 1;
+}
+
+.notion-eoi-position {
+  position: relative;
+}
+
+.notion-eoi-container {
+  width: 100%;
+  height: 100%;
+}
